message_auto_subscribe_notify_own/models/mail_thread.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# Copyright 2019 Aleph Objects, Inc.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    @api.multi
+    def message_auto_subscribe(self, updated_fields, values=None):
+        super(MailThread, self).message_auto_subscribe(updated_fields,
+                                                       values=values)
+        user_field_lst = self._message_get_auto_subscribe_fields(
+            updated_fields)
+        to_add_users = self.env['res.users'].sudo().browse(
```

Comment:
Tested on local. I created a lead and assign administrator as a sales person. Also tried to modified an existng lead and assign admin as sales person. I could not make it work. `user_field_lst` is always empty. Not sure what is wrong.